tiledb/sm/filter/filter_pipeline.h

```diff
@@ -255,7 +265,7 @@ class FilterPipeline {
   typedef std::pair<FilterBuffer, FilterBuffer> FilterBufferPair;
 
   /** The ordered list of filters comprising the pipeline. */
-  std::vector<tdb_unique_ptr<Filter>> filters_;
+  std::vector<std::shared_ptr<Filter>> filters_;
```

Comment:
Is there a particular reason to switch from unique to shared here?

Reply:
That is because FilterCreate::deserialize returns shared_ptr. and shared_ptr can not convert to uniqu_ptr(https://stackoverflow.com/questions/37884728/does-c11-unique-ptr-and-shared-ptr-able-to-convert-to-each-others-type)

Reply:
That is because FilterCreate::deserialize returns shared_ptr. and shared_ptr can not convert to uniqu_ptr(https://stackoverflow.com/questions/37884728/does-c11-unique-ptr-and-shared-ptr-able-to-convert-to-each-others-type)